src/main/java/org/truffleruby/core/fiber/FiberNodes.java

```diff
@@ -145,7 +145,7 @@ public static void start(RubyContext context, ThreadManager threadManager, Dynam
         threadManager.initializeValuesBasedOnCurrentJavaThread(Layouts.FIBER.getRubyThread(fiber), pThreadID);
 
         Layouts.THREAD.getFiberManager(Layouts.FIBER.getRubyThread(fiber)).registerFiber(fiber);
-        context.getSafepointManager().enterThread();
```

Comment:
The place to enterThread() and leaveThread() was carefully selected (e.g., the thread has a `pThreadID` before entering and knows its Java thread), so this change might introduce problems as they are now done around the rest of the initialization.

Reply:
We didn't merge this PR.